config/external_service.go

```diff
@@ -1,60 +1,136 @@
+	// except where there is a service-level override.
+	// If enabled is false, the global settings are not applied, only service-level settings (if declared).
+	Enabled bool `json:"enabled"`
+	// HTTPProxy is the HTTP proxy URL for HTTP requests to external services (e.g., http://localhost:3128).
+	// This setting applies globally unless overridden by service-specific configuration.
+	HTTPProxy string `json:"http_proxy"`
+	// HTTPSProxy is the HTTPS proxy URL for HTTPS requests to external services (e.g., http://localhost:3128).
+	// This setting applies globally unless overridden by service-specific configuration.
+	HTTPSProxy string `json:"https_proxy"`
+	// BypassProxy is a comma-separated list of hosts to bypass proxy. Supports exact hostnames, IP addresses,
+	// CIDR blocks, and wildcard patterns (e.g., localhost,127.0.0.1,.internal,*.local).
+	BypassProxy string `json:"bypass_proxy"`
+}
+
+// ProxyConfig defines service-specific HTTP proxy configuration.
 type ProxyConfig struct {
-	// UseEnvironment enables reading proxy configuration from environment variables
-	// (HTTP_PROXY, HTTPS_PROXY, NO_PROXY)
-	UseEnvironment bool `json:"use_environment"`
-	// HTTPProxy sets the HTTP proxy URL (e.g., http://proxy:8080)
+	// Enabled determines whether to apply proxy settings for this specific service.
+	Enabled bool `json:"enabled"`
+	// HTTPProxy is the HTTP proxy URL for HTTP requests (e.g., http://localhost:3128).
 	HTTPProxy string `json:"http_proxy"`
-	// HTTPSProxy sets the HTTPS proxy URL (e.g., https://proxy:8080)
+	// HTTPSProxy is the HTTPS proxy URL for HTTPS requests (e.g., http://localhost:3128).
 	HTTPSProxy string `json:"https_proxy"`
-	// NoProxy defines addresses that should bypass the proxy (comma-separated)
-	NoProxy string `json:"no_proxy"`
+	// BypassProxy is a comma-separated list of hosts to bypass proxy.
+	BypassProxy string `json:"bypass_proxy"`
 }
 
 // ServiceConfig defines service-specific configuration that can override global settings.
 type ServiceConfig struct {
-	// Proxy configuration for this specific service type
+	// Service-specific proxy configuration. When enabled, overrides global proxy settings for this service.
 	Proxy ProxyConfig `json:"proxy"`
-	// MTLS configuration for secure external communications
+	// Mutual TLS configuration for service communications. Supports certificates stored in Tyk Certificate Store
+	// (cert_id, ca_cert_ids) or in the file system (cert_file, key_file, ca_file).
+	// Tyk Certificate Store configuration takes priority if provided.
 	MTLS MTLSConfig `json:"mtls"`
 }
 
-// MTLSConfig defines mTLS client certificate configuration.
+// MTLSConfig defines mutual TLS client certificate configuration.
 type MTLSConfig struct {
-	// Enabled controls whether mTLS is enabled for this service
+	// Enabled controls whether mTLS is enabled for service requests. When enabled, cert_file and key_file are required.
```

Comment:
This should state that either File-based or Certificate Store configuration must be provided - it's currently misleading.